src/js/views/maps/legend/LegendContainerView.js

```diff
@@ -61,9 +62,49 @@ define([
         this.model.get("allLayers")?.forEach((layer) => {
           this.stopListening(layer, "change:visible");
           this.listenTo(layer, "change:visible", this.updateLegend);
+
+          // Updates the legend when the Filter model is updated (through the Filter by Property feature).
+          const colorPaletteProperty = layer
+            ?.get("colorPalette")
+            ?.get("property");
+          layer?.get("filters")?.forEach((model) => {
+            if (model.get("property") === colorPaletteProperty) {
+              this.filterModel = model;
+            }
+          });
+          if (this.filterModel) {
+            this.stopListening(this.filterModel, "change:values");
+            this.listenTo(this.filterModel, "change:values", this.updateLegend);
+          }
         });
       },
 
+      /**
+       * Sets the active state of the filter based values (for categorical
+       * filters) in the Color Palette model.
+       * @returns {AssetColorPalette} - Return the updated color palette model
+       * @param layer - Layer item model
+       * @param colorPaletteModel - Asset color palette model
+       */
+      updateColorPalette(layer, colorPaletteModel) {
+        const colorPaletteProperty = colorPaletteModel.get("property");
+        let layerValues;
+        layer.get("filters")?.forEach((model) => {
+          if (model.get("property") === colorPaletteProperty) {
+            layerValues = model.get("values");
```

Comment:
Could you just do:
`const layerValues = this.filterModel.get("values")`?

Because you already found the `filterModel` in `render()`?

Haven't tested, but seems like it would work?